solr/core/src/java/org/apache/solr/spelling/SpellCheckCollator.java

```diff
@@ -51,6 +51,14 @@ public class SpellCheckCollator {
   private boolean suggestionsMayOverlap = false;
   private int docCollectionLimit = 0;
 
+  private static volatile boolean mergeSpellCheckCorrections =
+      Boolean.parseBoolean(
+          System.getProperty("solr.SpellCheckCollator.mergeSpellCheckCorrections", "true"));
+
+  private static volatile boolean logSIOOBEDebugData =
```

Comment:
This seems like an unrelated change. 
Can we just add a debug line and switch the log level for the specific class instead of relying on a system property.

Reply:
sounds good @anshumg! removed that property and moved the entire things into a debug log.